### PR TITLE
feat(afterDeadline): ensure it's clear the seat is not exitable by a user before the deadline

### PR DESCRIPTION
**Pull request by 0xpatrickdev**
The exit behavior of an `afterDeadline` offer  _before_ the deadline occurs is not entirely clear in the documentation. The [current behavior](https://github.com/Agoric/agoric-sdk/blob/pismoD/packages/zoe/test/unitTests/zcf/test-zcf.js#L992-L994) is that it is in the `waived` state until the deadline occurs, but one could mistakenly think it is in the `onDemand` state until the deadline occurs without explicit mention.

This PR adds a sentence describing the full behavior to the **Proposals and payments** section. The second half is maybe not necessary, but [this comment](https://github.com/Agoric/agoric-sdk/blob/master/packages/zoe/src/contractFacet/exit.js#L70-L71) inspired me to include it.

It may also make sense to add this language or similar language to the other two places `afterDeadline` appears - Exit Rule in the glossary and the Zoe Guide.

### CODE REVIEW

**Review by ivanlei — APPROVED** [review]
LGTM

**Review by kbennett2000 — COMMENTED** [review]
Should more detail about the afterDeadline offer be included? Example:
"The exit behavior of an afterDeadline offer before the deadline occurs is that it is in the waived state until the deadline occurs; note that it is not in the onDemand state until the deadline occurs."

File: main/reference/zoe-api/zoe.md [hunk 1]
```diff
@@ -314,6 +314,7 @@ key:value pairs:
   by its **timer** and **deadline** properties.
   **timer** must be a timer, and **deadline** must be a [BigInt](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt) value interpreted with respect to the timer.
   Some example timers use Unix epoch time, while others count block height.
+  The proposer cannot exit the seat before the deadline, but the contract can exit the seat early on their behalf.
```
Comment (0xpatrickdev) [inline main/reference/zoe-api/zoe.md:317]:
```suggestion
  The proposer cannot exit the seat before the deadline - the seat is in the `waived` state until the deadline, but the contract can exit the seat early on the proposer's behalf.
```

Comment (0xpatrickdev) [reply main/reference/zoe-api/zoe.md:317]: Realizing the section this is proposing to edit is no longer there. Will update this to target the https://docs.agoric.com/reference/zoe-api/zoe.html#proposals section.

**Comment by 0xpatrickdev** [comment]
Closing in favor of #969

**Pull request closed without merging**